### PR TITLE
Enable comments on pdfs 

### PR DESCRIPTION
**Pull request by JanAckermann**
Before this PR ist was not possible to add comments on PDF files with rich documents/Collabora, this works now.

Fixes: - https://github.com/owncloud/enterprise/issues/4606

Before:
image

After:
image

### CODE REVIEW

**Comment by JanAckermann** [comment]
https://github.com/CollaboraOnline/online/commit/ca00470722ec7a7a7b190ca36a1f71da06ed6847

File: lib/Controller/DocumentController.php [hunk 1]
```diff
@@ -426,7 +426,7 @@ private function handleDocIndex($fileId, $shareToken, $currentUser) {
 
 		// Update permissions
 		$permissions = $doc['permissions'];
-		if (!($doc['action'] === 'edit')) {
+		if (!($doc['action'] === 'edit') && !($doc['action'] === 'view_comment')) {
```
Comment (mrow4a) [inline lib/Controller/DocumentController.php:429]: it is **&&** or **||** ? hm

Comment (JammingBen) [reply lib/Controller/DocumentController.php:429]: `&&` is correct 🙂 

The action for PDFs is `view_comment`, and we want to include the update permission in this case. With `||` we would run into the if-block which would remove this permission.

Comment (JanAckermann) [reply lib/Controller/DocumentController.php:429]: As @JammingBen pointed out, the line:
`$permissions & ~\OCP\Constants::PERMISSION_UPDATE;`
removes the PERMISSION_UPDATE from the pemissions var

**Review by jvillafanez — APPROVED** [review]

**Pull request merged**